### PR TITLE
Move synopsis options from meta index to index

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

In preparation for splitting out the meta index into a separate actor, this moves the synopsis factory options to the index directly.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

### CODE REVIEW

**Review by lava — APPROVED** [review]
Also looks good to me.

File: libvast/vast/meta_index.hpp [hunk 1]
```diff
@@ -73,16 +73,12 @@ class meta_index {
...
-  /// @returns A reference to the synopsis options.
-  caf::settings& factory_options();
-
   // -- concepts ---------------------------------------------------------------
 
   // Allow debug printing meta_index instances.
   template <class Inspector>
   friend auto inspect(Inspector& f, meta_index& x) {
-    return f(x.synopsis_options_, x.synopses_);
+    return f(x.synopses_);
```
Comment (tobim) [inline libvast/vast/meta_index.hpp:81]: I would suggest to insert a dummy `caf::settings` value to avoid the unnecessary breaking.

Comment (dominiklohmann) [reply libvast/vast/meta_index.hpp:81]: We still serialize this with CAF? 🙄

Comment (tobim) [reply libvast/vast/meta_index.hpp:81]: hm, scratch that, we're not writing this to file.

Comment (dominiklohmann) [reply libvast/vast/meta_index.hpp:81]: Alrighty then.

File: libvast/test/meta_index.cpp [hunk 2]
```diff
@@ -121,7 +121,7 @@ struct fixture {
     for (size_t i = 0; i < num_partitions; ++i) {
       auto name = i % 2 == 0 ? "foo"s : "foobar"s;
       auto& part = mock_partitions.emplace_back(std::move(name), ids[i], i);
-      auto ps = make_partition_synopsis(part.slice, meta_idx.factory_options());
+      auto ps = make_partition_synopsis(part.slice, {});
```
Comment (lava) [inline libvast/test/meta_index.cpp:124]: You can also remove the second argument, this is a helper function just for this unit test.

Comment (dominiklohmann) [reply libvast/test/meta_index.cpp:124]: Thanks, totally missed that.

**Pull request merged**